spatialdata/_core/_spatialdata.py

```diff
@@ -242,12 +280,391 @@ def _init_add_element(self, name: str, element_type: str, overwrite: bool) -> za
+                except np.linalg.LinAlgError:
+                    pass
+        return g
+
+    def map_coordinate_systems(
+        self,
+        source_coordinate_system: Union[SpatialElement, str],
+        target_coordinate_system: Union[SpatialElement, str],
+        intermediate_coordinate_systems: Optional[Union[SpatialElement, str]] = None,
+    ) -> BaseTransformation:
+        """
+        Get the transformation to map a coordinate system (intrinsic or extrinsic) to another one.
+
+        Parameters
+        ----------
+        source_coordinate_system
+            The source coordinate system. Can be a SpatialElement (intrinsic coordinate system) or a string (extrinsic
+            coordinate system).
+        target_coordinate_system
+            The target coordinate system. Can be a SpatialElement (intrinsic coordinate system) or a string (extrinsic
+            coordinate system).
+
+        Returns
+        -------
+        The transformation to map the source coordinate system to the target coordinate system.
+        """
+        if (
+            isinstance(source_coordinate_system, str)
+            and isinstance(target_coordinate_system, str)
+            and source_coordinate_system == target_coordinate_system
+            or id(source_coordinate_system) == id(target_coordinate_system)
+        ):
+            return Identity()
+        else:
+            g = self._build_transformations_graph()
+            src_node: Union[int, str]
+            if has_type_spatial_element(source_coordinate_system):
+                src_node = id(source_coordinate_system)
+            else:
+                src_node = source_coordinate_system
+            des_node: Union[int, str]
+            if has_type_spatial_element(target_coordinate_system):
+                des_node = id(target_coordinate_system)
+            else:
+                des_node = target_coordinate_system
+            paths = list(nx.all_simple_paths(g, source=src_node, target=des_node))
+            if len(paths) == 0:
+                # error 0 (we refer to this in the tests)
+                raise RuntimeError("No path found between the two coordinate systems")
+            elif len(paths) > 1:
+                if intermediate_coordinate_systems is None:
+                    # if one and only one of the paths has lenght 1, we choose it straight away, otherwise we raise
+                    # an expection and ask the user to be more specific
+                    paths_with_length_1 = [p for p in paths if len(p) == 2]
+                    if len(paths_with_length_1) == 1:
+                        path = paths_with_length_1[0]
+                    else:
+                        # error 1
+                        raise RuntimeError(
+                            "Multiple paths found between the two coordinate systems. Please specify an intermediate "
+                            "coordinate system."
+                        )
```

Comment:
maybe it's worth here to print the path to tie it to what went wrong?

Reply:
Good point, added it. Now the exception looks like this:
```
>                       raise RuntimeError(
                            "Multiple paths found between the two coordinate systems. Please specify an intermediate "
                            f"coordinate system. Available paths are:{s}"
                        )
E                       RuntimeError: Multiple paths found between the two coordinate systems. Please specify an intermediate coordinate system. Available paths are:
E                           my_space0 -> /images/image2d_multiscale -> global
E                           my_space0 -> /labels/labels2d -> global
```

Reply:
Actually I changed it to this so it's clearer what the user should plug in the function call:
```
E                       RuntimeError: Multiple paths found between the two coordinate systems. Please specify an intermediate coordinate system. Available paths are:
E                           my_space0 -> <sdata>.images['image2d_multiscale'] -> global
E                           my_space0 -> <sdata>.labels['labels2d'] -> global
```